### PR TITLE
#164858950 Query office structures v2

### PR DESCRIPTION
**Pull request by rajeman**
**What does this PR do?**

Refactor query for office structures

**Description of the task to be completed?**

- A user should be able to query for all office structures
- A user should be able to query for a structure by supplying its webId

**How should this be manually tested?**

- use the query below to get all office structures
```query {
  allStructures {
    name
    tag
    locationId
  }
```
- use the query below to get a single office structure

```query {
  structureByStructureId(webId: "some structureId") {
    name
    tag
    locationId
  }
```

**Any background context you want to provide?**

N/A

**What are the relevant pivotal tracker stories?**

[#164858950](https://www.pivotaltracker.com/n/projects/2154921)

Checklist:
 - [x] My code follows the style guidelines of this project
 - [x] I have linted my code prior to submission
 - [x] Implementation works according to expectations
 - [x] My changes generate no new warnings
 - [x] Existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by dnuwa** [comment]
@rajeman your commit message needs improvement. kindly check out the example in the engineering playbook. `https://github.com/andela/engineering-playbook/tree/master/5.%20Developing/Conventions#commit-message`
image

File: tests/test_structure/test_all_structures.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from tests.base import BaseTestCase, CommonTestCases
+from fixtures.structure.structures_fixtures import (
+    query_structures,
+    expected_response_structures
+)
+
+
+class TestAllStructures(BaseTestCase):
+
+    def test_all_structures(self):
```
Comment (dnuwa) [inline tests/test_structure/test_all_structures.py]: this is good work @rajeman though you need to add docstrings to all your tests.

File: fixtures/structure/structures_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        name
+        parentId
+        tag
+        locationId
+        position
+      }
+    }
+        '''
+
+expected_response_structures = {
```
Comment (ThaDeveloper) [inline fixtures/structure/structures_fixtures.py]: @rajeman `structures_query_response` or `expected_structures_query_response` would be more descriptive.

**Comment by rajeman** [comment]
@ThaDeveloper @dnuwa 
I have made the changes

**Comment by KMaina** [comment]
Good work @rajeman, I was able to retrieve office structures created locally.

File: api/structure/schema.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class Structure(SQLAlchemyObjectType):
+    class Meta:
+        model = StructureModel
+
+
+class Query(graphene.ObjectType):
+    all_structures = graphene.List(Structure)
+    get_structure_by_web_id = graphene.Field(Structure,
```
Comment (Bonifase) [inline api/structure/schema.py]: Variable names are not supposed to be action in nature, use nouns

File: tests/test_structure/test_structure.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        """
+        CommonTestCases.admin_token_assert_equal(
+            self,
+            structure_query,
+            expected_structure_query_response
+        )
+
+    def test_structure_non_existant_web_id(self):
+        """
+        Test for office structure with non existant webId
```
Comment (Bonifase) [inline tests/test_structure/test_structure.py]: What are we testing here? Please be specific in your description. Same to in line 37 below

**Comment by rajeman** [comment]
@Bonifase I have made the requested changes

File: api/structure/schema.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+from helpers.auth.authentication import Auth
+from api.structure.models import Structure as StructureModel
+
+
+class Structure(SQLAlchemyObjectType):
+    class Meta:
+        model = StructureModel
+
+
+class Query(graphene.ObjectType):
```
Comment (TeamoreA) [inline api/structure/schema.py]: Good work @rajeman .It would be better if you added descriptions for these two queries.

Comment (rajeman) [reply api/structure/schema.py]: @TeamoreA I have done that

File: api/structure/schema.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    """
+      Query for office structures
+    """
+    all_structures = graphene.List(Structure)
```
Comment (RachaelNantale) [inline api/structure/schema.py]: hey @rajeman, Great work so far. Please add the descriptions of the different queries. if help is needed, then reach out to me

File: api/structure/schema.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    structure_by_web_id = graphene.Field(Structure, web_id=graphene.String())
+
+    @Auth.user_roles('Admin')
+    def resolve_all_structures(self, info):
+        """
+          Returns a list containing all office structures
+        """
+        query = Structure.get_query(info)
```
Comment (rajeman) [inline api/structure/schema.py]: @RachaelNantale  are you talking about this?

Comment (RachaelNantale) [reply api/structure/schema.py]: `all_structures = graphene.List(Structure, description ="description goes here")`
Try something like this

**Comment by dnuwa** [comment]
good work @rajeman please always endeavor to respond to reviews so the team would know you have implemented the changes.

File: api/structure/schema.py (hunk 7 above)
Comment (rajeman) [reply api/structure/schema.py]: @RachaelNantale I have implemented the change

File: api/structure/schema.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      Structure, web_id=graphene.String(), description="Returns the office \
+        structure corresponding to the provided webId")
+
+    @Auth.user_roles('Admin')
+    def resolve_all_structures(self, info):
+        query = Structure.get_query(info)
+        return query.order_by(func.lower(StructureModel.name)).all()
+
+    @Auth.user_roles('Admin')
+    def resolve_structure_by_web_id(self, info, web_id):
```
Comment (Bonifase) [inline api/structure/schema.py]: Use structure_id instead of web_id

**Comment by rajeman** [comment]
@Bonifase Requested changes have been made. Kindly review

File: tests/test_structure/test_all_structures.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
+from tests.base import BaseTestCase, CommonTestCases
+from fixtures.structure.structures_fixtures import (
+    structures_query,
+    expected_structures_query_response
+)
+
+
+class TestAllStructures(BaseTestCase):
+
+    def test_all_structures(self):
```
Comment (Bonifase) [inline tests/test_structure/test_all_structures.py]: Rename this to `test_query_all_structures_works`, same to other tests. Function names should be descriptive enough to encourage readability.

Comment (rajeman) [reply tests/test_structure/test_all_structures.py]: @Bonifase changes have been made

**Comment by rajeman** [comment]
@Bonifase conflicts resolved

**Pull request merged**